aten/src/ATen/native/utils/ParamsHash.h

```diff
@@ -38,5 +38,57 @@ struct ParamsEqual {
   }
 };
 
+// Provide explicit byte-for-byte constructors to avoid uwittingly leaving
+// padding bytes unitialized (e.g., when passing Params by value)
+template <typename T>
+struct ParamsWrapper {
```

Comment:
Hmm, indeed, does `__attribute__((packed))` help? If we make the cache keys packed, then there will be no issue of uninitialized bytes?

Reply:
We cannot just overload constructors directly because that would violate the POD requirement (of the original `ParamsHash`.

`__attribute__((packed))` could work, but I think there are concerns about it being system dependent (unsafe on some systems).

Reply:
Is the original `ParamsHash` still used?

Reply:
It is still used in v7 and I believe some other backends. If this approach is looks OK as an replacement I can remove it in cleanup/follow up PRs.